### PR TITLE
Define ETCD Transaction limit

### PR DESCRIPTION
**Pull request by berndverst**
# Description

Define ETCD Transaction limit

Because the transaction limit in ETCD depends on the server (via server flag `--max-txn-ops`), but with a default of 128 if unset I have also exposed this configuration option via a new `maxTxnOps` component metadata option.

### CODE REVIEW

File: state/etcd/etcd.go [hunk 1]
```diff
@@ -53,6 +54,8 @@ type etcdConfig struct {
 	CA        string `json:"ca"`
 	Cert      string `json:"cert"`
 	Key       string `json:"key"`
+	// Transaction server options
```
Comment (ItalyPaleAle) [inline state/etcd/etcd.go:57]: Nit:

```suggestion
	// Transactional store options
```

File: state/etcd/etcd.go [hunk 2]
```diff
@@ -134,7 +138,10 @@ func (e *Etcd) Features() []state.Feature {
 }
 
 func metadataToConfig(connInfo map[string]string) (*etcdConfig, error) {
-	m := &etcdConfig{}
+	m := &etcdConfig{
+		// This is the default value for maximum ops per transaction, configurtable via etcd server flag --max-txn-ops.
+		MaxTxnOps: 128,
```
Comment (ItalyPaleAle) [inline state/etcd/etcd.go:143]: Can you make this a constant?

File: state/etcd/etcd.go (hunk 1 above)
Comment (berndverst) [reply state/etcd/etcd.go:57]: Not what I was going for. I want to be clear that these options must match options configured on the server. Other suggestions @ItalyPaleAle ?

File: state/etcd/etcd.go (hunk 2 above)
Comment (berndverst) [reply state/etcd/etcd.go:143]: I guess - but this right here is how we initialize defaults other places too. I can point you to many examples in contrib.

File: state/etcd/etcd.go (hunk 1 above)
Comment (ItalyPaleAle) [reply state/etcd/etcd.go:57]: It was just a nit.

File: state/etcd/etcd.go (hunk 2 above)
Comment (ItalyPaleAle) [reply state/etcd/etcd.go:143]: Ok. It was just another nit too.

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**